etcdmain/gateway.go

```diff
@@ -81,6 +85,37 @@ func startGateway(cmd *cobra.Command, args []string) {
 			os.Exit(1)
 		}
 		plog.Infof("discovered the cluster %s from %s", eps, gatewayDNSCluster)
+
+		// confirm TLS connections are good
+		tlsInfo := transport.TLSInfo{
```

Comment:
would `tlsutil` be better? not sure about how to draw the lines between transport/tlsutil/netutil